packages/bpk-component-tooltip/src/BpkTooltip.tsx

```diff
@@ -22,33 +22,122 @@ This is the component for the tooltip that is displayed to users.
 The actual component that developers create (i.e. the default export from this package) is BpkTooltipPortal.
 */
 
-import type { ReactNode } from 'react';
+import type { ReactNode, ReactElement } from 'react';
+import { cloneElement, useRef, useState } from 'react';
+
+import {
+  arrow,
+  flip,
+  FloatingArrow,
+  FloatingPortal,
+  offset,
+  shift,
+  useFloating,
+  useHover,
+  useInteractions,
+  useRole,
+} from '@floating-ui/react';
+
+import { surfaceHighlightDay } from '@skyscanner/bpk-foundations-web/tokens/base.es6';
 
 import { TransitionInitialMount, cssModules } from '../../bpk-react-utils';
 
 import { ARROW_ID, TOOLTIP_TYPES } from './constants';
 
+import type { Placement } from '@floating-ui/react';
+
 import STYLES from './BpkTooltip.module.scss';
 
 const getClassName = cssModules(STYLES);
 
-export type TooltipProps = {
+// The stroke width is used to set the border width of the arrow.
+const strokeWidth = 1;
+
+export type Props = {
+  /**
+   * Tooltips are invisible to assistive technologies such as screen readers.
+   * To improve accessibility, `ariaLabel` is required to describe the content of the tooltip to assistive technologies.
+   * The label will be used on the `target` element, so any existing `aria-label` attached to `target` will be overridden.
+   */
+  ariaLabel: string;
+  /**
+   * "target" should be a DOM element with a "ref" attached to it.
```

Comment:
is that still the case?

Reply:
Yes this is still the case, we will still want people to pass through a ReactElement and not a function like `document.getElementById` which is what the popover did, but the tooltip was never built to support that only 

```
<div>
   My target
</div>
```

Reply:
I mean the `ref` part `with a "ref" attached to it.` . I see we're using floating ui to set the ref so I don't think the consumer has to? https://github.com/Skyscanner/backpack/pull/3485/files#diff-2702cb346943a8761bb30c82e1a61efeae40297ed8cee8028da4376b719dc73cR132-R138

Also interesting that you can clone an element and attach a ref to it 😮 I think that's new-ish, I think a few years ago it was not possible

Reply:
Ah gotcha! Let me do a test with removing that target ref and see what magic happens!

> Also interesting that you can clone an element and attach a ref to it 😮 I think that's new-ish, I think a few years ago it was not possible

Interesting didn't realise it was a newer feature! I guess it is just a property to a DOM element so is easy to add this prop as any other?

Reply:
> Ah gotcha! Let me do a test with removing that target ref and see what magic happens!

Cool, have you checked that?

> Interesting didn't realise it was a newer feature! I guess it is just a property to a DOM element so is easy to add this prop as any other?

Yeah, I remember it wasn't possible before, but not sure why 🤔